csi/volumeutils.py

```diff
@@ -855,6 +878,51 @@ def generate_client_volfile(volname):
         content = template_file.read()
 
     Template(content).stream(**data).dump(client_volfile)
+    return True
+
+def reload_glusterfs(volume):
+    """Mount Glusterfs Volume"""
+    if volume["type"] == "External":
+        return False
+
+    volname = volume["name"]
+
+    if not VOL_DATA.get(volname, None):
+        return False
+
+    # Ignore if already glusterfs process running for that volume
+    with mount_lock:
+        if not generate_client_volfile(volname):
+            return False
+        # TODO: ideally, keep the pid in structure for easier access
+        # pid = VOL_DATA[volname]["pid"]
+        # cmd = ["kill", "-HUP", str(pid)]
+        cmd = ["ps", "--no-header", "-ww", "-o", "pid,command", "-C", "glusterfs"]
+
+        try:
+            out, err, _ = execute(*cmd)
+            for line in out.split("\n"):
+                parts = line.split()
+                pid = parts[0]
+                for part in parts:
+                    if part.startswith("--volume-id="):
+                        if part.split("=")[-1] == volname:
+                            cmd1 = [ "kill", "-HUP", pid ]
+                            execute(*cmd1)
```

Comment:
Move this to a function, so that early return(loop break) after sighup execution.

Reply:
Done in #570